File: Tema 4. Modelo de Programación Lógica/AA 4.1 Reporte práctico/sistema_riego.pl
% Se tendrán las zonas norte, sur y centro
% Hechos
humedad(norte, baja).
humedad(sur, baja).
humedad(centro, baja).

temperatura(norte, 34).
temperatura(sur, 30).
temperatura(centro, 36).

pronostico_lluvia(norte, false).
pronostico_lluvia(sur, false).
pronostico_lluvia(centro, true).

hora(19).

% Reglas
hora_adecuada :-   hora(H), (H < 10 ; H > 18).

regar(Zona) :- 
    humedad(Zona, baja),
    pronostico_lluvia(Zona, false),
    hora_adecuada.

estado(Zona, 'Activado') :- regar(Zona), !.
estado(Zona, 'No activado') :- \+ regar(Zona).

alerta(Zona) :-    temperatura(Zona, T), T >= 32.

recomendacion(Zona) :-
    regar(Zona),
    alerta(Zona), !,
    writeln('ALERTA: Riego activado con temperatura alta. Considere regar de noche o por goteo').

recomendacion(Zona) :-
    \+ regar(Zona), !,
    writeln('No es necesario activar el riego. Está lloviendo').

recomendacion(Zona) :- writeln(['Sin recomendaciones especiales para el riego', Zona]).

File: Tema 4. Modelo de Programación Lógica/AA 4.1 Reporte práctico/AA 4.2 Reporte práctico/sistema_riego.py
#Datos para cada zona
humedad = {
    'norte': 'baja',
    'sur': 'baja',
    'centro': 'baja'
}

temperatura = {
    'norte': 34,
    'sur': 30,
    'centro': 36
}

pronostico_lluvia = {
    'norte': False,
    'sur': False,
    'centro': True
}

hora = 19

#Funciones del sistema
def hora_adecuada(hora):
    return hora < 10 or hora > 18

def regar(zona):
    if humedad[zona]=='baja' and pronostico_lluvia[zona]==False and hora_adecuada(hora):
        return True
    return False

def estado(zona):
    if regar(zona):
        return 'Activado'
    else:
        return 'No activado'

def alerta(zona):
    return temperatura[zona] >= 32

def recomendacion(zona):
    if regar(zona) and alerta(zona):
        return 'ALERTA: Riego activado con temperatura alta. Considere regar de noche o por goteo'
    elif regar(zona):
        return 'Sin recomendaciones especiales para el riego'
    else:
        return 'No es necesario activar el riego. Está lloviendo'

def mostrarInfo():
    for zona in humedad:
        print(f"Zona: {zona}")
        print(f"Estado: {estado(zona)}")
        print(f"Recomendación: {recomendacion(zona)}")
        print()

mostrarInfo()